usecases/internal/measurement.go

```
@@ -62,16 +72,401 @@ func MeasurementPhaseSpecificDataForFilter(
+//
+// Example:
+//
+//	measurements := []model.MeasurementDataType{...}
+//	value, err := GetMeasurementValue(measurements, PowerMeasurementValidator)
+//	if err != nil {
+//		return 0, api.ErrDataNotAvailable
+//	}
+//	return value, nil
+func GetMeasurementValue(measurements []model.MeasurementDataType, validator *MeasurementValidator) (float64, error) {
+	// Convert slice to pointer slice for validation
+	ptrMeasurements := make([]*model.MeasurementDataType, len(measurements))
+	for i := range measurements {
+		ptrMeasurements[i] = &measurements[i]
+	}
+	
+	valid, err := validator.ValidateFirst(ptrMeasurements)
+	if err != nil {
+		return 0, api.ErrDataNotAvailable
+	}
+	
+	if valid == nil || valid.Value == nil {
+		return 0, api.ErrDataNotAvailable
+	}
+	
+	return valid.Value.GetValue(), nil
+}
+
+// Measurement-specific validation rules
+
+// RequireMeasurementId ensures MeasurementId is present.
+//
+// This is the most basic measurement validation rule.
+// Almost all measurement validators should include this rule.
+func RequireMeasurementId() ValidationRule[*model.MeasurementDataType] {
+	return RequireField(
+		func(m *model.MeasurementDataType) *model.MeasurementIdType { return m.MeasurementId },
+		"MeasurementId",
+	)
+}
+
+// RequireMeasurementValue ensures Value is present.
+//
+// This rule ensures that measurement data contains an actual value.
+// Use this when you need to extract numeric values from measurements.
+func RequireMeasurementValue() ValidationRule[*model.MeasurementDataType] {
+	return RequireScaledNumber(
+		func(m *model.MeasurementDataType) *model.ScaledNumberType { return m.Value },
+		"Value",
+	)
+}
+
+// RequireValueType ensures ValueType matches expected type
+func RequireValueType(expected model.MeasurementValueTypeType) ValidationRule[*model.MeasurementDataType] {
+	return func(m *model.MeasurementDataType) error {
+		if expected == "" {
+			return nil // Skip if no expected type specified
+		}
```

This feels like a caller error that we shouldn't be silently ignoring here, if somebody calls this function incorrectly and it isn't doing what they're expecting they should get an error